// containers/phone/index.js
import React, {Component} from "react";
import {connect} from 'react-redux';
import  {fetchPhoneById, addPhoneToBasket} from 'actions/index';
import {getPhoneById} from 'selectors';
import Nav from 'component/navigation'

import './phone.scss'
import {Link} from "react-router-dom";

class Phone extends Component{

    renderFields = () => {
        const {phone} = this.props;
        let newObj = {
            cpu: phone.cpu,
            camera: phone.camera,
            size: phone.size,
            weight: phone.weight,
            display: phone.display,
            battery: phone.battery,
            memory: phone.memory
        };
        newObj = Object.entries(newObj);
        return newObj.map(([key,value]) =>
            <div className='column' key={key}>
                <div className='ab-details-title'>
                    <p>{key}</p>
                </div>
                <div className='ab-details-info'>
                    <p>{value}</p>
                </div>
            </div>
        )
    };

    componentDidMount() {
        this.props.fetchPhoneById(this.props.match.params.id)
    }

    renderContent = () => {
        const {phone} = this.props;
        return(
            <div className='thumbnail'>
                <div className='row'>
                    <div className='col-md-6'>
                        <img
                            className='img-card'
                            src={phone.image}
                            alt={phone.name}
                        />
                    </div>
                    <div className='col-md-6'>
                        {this.renderFields()}
                    </div>
                </div>
                <div className='caption-full'>
                    <h4>{phone.name}</h4>
                    <h4 className='pull-right'>${phone.price}</h4>
                    <p>{phone.description}</p>
                </div>
            </div>
        )
    };

    renderSidebar = () => {
        const {phone, addPhoneToBasket} = this.props;
        console.log(this.props)
        return(
            <div className='sidebar'>
                <div className='form-group'>
                    <h3>{phone.name}</h3>
                    <h4>${phone.price}</h4>
                </div>
                <Link to='/' className='btn btn-basket-static'>
                    Back to store
                </Link>
                <button
                    type='button'
                    className='btn btn-success btn-block'
                    onClick={() => addPhoneToBasket(phone.id)}
                >
                    Add to cart
                </button>
            </div>
        )
    };

    render() {
        const {phone} = this.props;
        return (
            <div className='view-container single-Mob'>
                <Nav/>
                <div className='container'>
                    <div className='row'>
                        <div className='col-md-3 mt-4'>
                            {phone && this.renderSidebar()}
                        </div>
                        <div className='col-md-9 mt-4'>
                            {phone && this.renderContent()}
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = state => ({
    phone: getPhoneById(state, state.phonePage.id)
});

const mapDispatchToProps = {
    fetchPhoneById,
    addPhoneToBasket
};

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Phone)
