// actions/index.js
import * as actionType from 'actionTypes';
import {fetchPhonesApi, fetchMorePhonesApi, fetchPhoneByIdApi, fetchCategoriesApi, editPhoneByIdApi} from 'api'
import {getRenderedPhonesLength} from "../selectors";

export const fetchPhones = () =>  async dispatch => {
    dispatch({
        type: actionType.FETCH_PHONES_START
    });

    try {
        const phones = await fetchPhonesApi();
        dispatch({
            type: actionType.FETCH_PHONES_SUCCESS,
            payload: phones
        })
    } catch (err) {
        dispatch({
            type: actionType.FETCH_PHONES_FAILURE,
            payload: err,
            error: true
        })
    }
};

export const loadMorePhones = () =>  async (dispatch, getState) => {
    const offset = getRenderedPhonesLength(getState());
    dispatch({
        type: actionType.LOAD_MORE_PHONES_START
    });

    try {
        const phones = await fetchMorePhonesApi({offset});
        dispatch({
            type: actionType.LOAD_MORE_PHONES_SUCCESS,
            payload: phones
        })
    } catch (err) {
        dispatch({
            type: actionType.LOAD_MORE_PHONES_FAILURE,
            payload: err,
            error: true
        })
    }
};

export const fetchPhoneById = id => async dispatch => {
    dispatch({
        type: actionType.FETCH_PHONE_BY_ID_START
    });

    try{
        const phone = await fetchPhoneByIdApi(id);
        dispatch({
            type: actionType.FETCH_PHONE_BY_ID_SUCCESS,
            payload: phone
        })
    }catch (err) {
        dispatch({
            type: actionType.FETCH_PHONE_BY_ID_FAILURE,
            payload: err,
            error: true
        })
    }
};

export const addPhoneToBasket = id => dispatch => {
    dispatch({
        type: actionType.ADD_PHONE_TO_BASKET,
        payload: id
    })
};

export const searchPhone = text => dispatch => {
    dispatch({
        type: actionType.SEARCH_PHONE,
        payload: text
    })
};

export const fetchCategories = () =>  async dispatch => {
    dispatch({
        type: actionType.FETCH_CATEGORIES_START
    });

    try {
        const categories = await fetchCategoriesApi();
        dispatch({
            type: actionType.FETCH_CATEGORIES_SUCCESS,
            payload: categories
        })
    } catch (err) {
        dispatch({
            type: actionType.FETCH_PHONE_BY_ID_FAILURE,
            payload: err,
            error: true
        })
    }
};

export const removePhoneFromBasket = id => dispatch => {
    dispatch({
        type: actionType.REMOVE_PHONE_FROM_BASKET,
        payload: id
    })
};

export const cleanBasket = () => dispatch => {
    dispatch({
        type: actionType.CLEAN_BASKET,
    })
};

export const basketCheckout = phones => {
    alert(JSON.stringify(phones))
};

export const editPhoneId = (id, data) => async dispatch => {
    dispatch({
        type: actionType.GET_EDIT_PHONE_BY_ID,
    });
    try {
        await editPhoneByIdApi(id, data);
        dispatch({
            type: actionType.GET_EDIT_PHONE_BY_ID_SUCCESS,
            payload: id,
            data: data
        })
    } catch (err) {
        dispatch({
            type: actionType.GET_EDIT_PHONE_BY_ID_FAILURE,
            payload: err,
            error: true
        })
    }
};
