// component/modal/index.js
import React from "react";
import {Modal as ModalANTD, Input} from 'antd';
import {getEditPhone} from "../../selectors";
import {connect} from "react-redux";
import {fetchPhones} from "actions";

class Modal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: '',
            id: '',
            cpu: '',
            camera: '',
            size: '',
            weight:  '',
            display: '',
            battery: '',
            memory: '',
            description:'',
            price: 100,
        }
    }

    componentDidUpdate(prevProps, prevState) {
        const {editOnePhone = {}} = this.props;
        const {editOnePhone: editOnePhonePrev = {}} = prevProps;
        if (editOnePhone.id !== editOnePhonePrev.id) {
            this.setState({
                ...editOnePhone,
            });
        }
    }

    toggleEdit = () => {
        const {onOk, editPhoneId} = this.props;
        editPhoneId(this.state.id, this.state);
        onOk();
    };

    handleChangeEvent = (e) => {
        this.setState({
            [e.target.name]: e.target.value
        });
    };

    renderFields = () => {
        return Object.keys(this.state).map((key) =>
             <div className='column' key={key}>
                 <div className='phone-title'>
                     <p>{key}</p>
                 </div>
                 <div className='phone-info'>
                     <Input
                         type="text"
                         value={this.state[key]}
                         onChange={this.handleChangeEvent}
                         name={key}
                     />
                 </div>
             </div>
         )
    };

    render() {
        return (
            <ModalANTD
                title="Edit Phone"
                visible={this.props.visible}
                onOk={() => this.toggleEdit()}
                onCancel={this.props.onCancel}
            >
                <form>
                    <div className='inputWrapper'>
                        {this.renderFields()}
                    </div>
                </form>
            </ModalANTD>
        )
    }
}
const mapStateToProps = (state, ownProps) => ({
    editOnePhone: getEditPhone(state, ownProps)
});

const mapDispatchToProps = {
    fetchPhones
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Modal)
