// containers/admin/addPhone/index.js
import React from "react";
import {Input, Button} from "antd";
import {Link} from "react-router-dom";
import { CheckOutlined } from '@ant-design/icons';

class PhoneAddBlock extends React.Component {
    renderFields = () => {
        let newObj = {
            name: '',
            cpu: '',
            camera: '',
            size: '',
            weight: '',
            display: '',
            battery: '',
            memory: '',
            description: '',
            price: 500
        };

        const handleInputChange = (e) => {
        };

        newObj = Object.entries(newObj);

        return newObj.map(([key, value]) =>
            <div className='column col-6' key={key}>
                <div className='phone-title'>
                    <p>{key}</p>
                </div>
                <div className='phone-info'>
                    <Input
                        type="text"
                        value={value}
                        onChange={handleInputChange}
                        name={key}
                    />
                </div>
            </div>
        )
    };
    render() {
        return (
            <div className='container'>
                <div className='row'>
                    <div className="col-6 mt-5">
                        <Link
                            to='/admin'
                            className='btn btn-basket-static'
                        >
                            <span>Back to Admin Panel</span>
                        </Link>
                    </div>
                    <div className='col-12 mt-5'>
                        <h5 className='mb-5'>Add New Phone</h5>
                        <form>
                            <div className='inputWrapper'>
                                {this.renderFields()}
                            </div>
                        </form>
                        <div className="row">
                            <div className="col-6 mt-5 mb-5">
                                <Button className='d-inline-flex align-items-center justify-content-center' type="primary" block>
                                    finish
                                    <CheckOutlined />
                                </Button>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

export default PhoneAddBlock
