// containers/admin/index.js
import React, {Component} from "react";
import { connect } from 'react-redux'
import {fetchPhones, editPhoneId} from 'actions';
import {getPhones} from "../../selectors";
import {Link} from "react-router-dom";
import Modal from 'component/modal'
import {Button} from 'antd';
import 'antd/dist/antd.css';
import { EditOutlined, DeleteOutlined} from '@ant-design/icons';

class Admin extends Component {
    constructor(props){
        super(props);
        this.state = {
            modalEditVisible: false,
        }
    }

    showModalEdit = (id) => {
        this.props.editPhoneId(id);
        this.setState({
            modalEditVisible: true,
        });
    };

    setModalEditVisible(modalEditVisible) {
        this.setState({ modalEditVisible });
    }

    componentDidMount() {
        this.props.fetchPhones();
    }

    renderPhone = (phone, index) => {
        return (
            <tr
                key={index}
                className='item-checout'
                id={phone.id}
            >
                <td className='first-column-checkout'>
                    <img
                        className='table-basket-img'
                        src={phone.image}
                        alt={phone.name}
                    />
                </td>
                <td>{phone.name}</td>
                <td>
                    <Button onClick={() => this.showModalEdit(phone.id)} className='d-inline-flex align-items-center' type='primary'>
                        edit
                        <EditOutlined/>
                    </Button>
                </td>
                <td>
                    <Button className='d-inline-flex align-items-center' type='danger'>
                        delete
                        <DeleteOutlined/>
                    </Button>
                </td>
            </tr>
        )
    };

    render() {
        const {phones, editPhoneId} = this.props;
        return (
            <div className='view-container'>
                <div className='container'>
                    <div className='row'>
                        <div className="col-6 mt-5">
                            <Link
                                to='/'
                                className='btn btn-basket-static'
                            >
                                <span>Back to store</span>
                            </Link>
                            <Link
                                to='/admin/addPhone'
                                className='btn btn-basket-static'
                            >
                                <span>Add Phone</span>
                            </Link>
                        </div>
                        <div className='table-responsive col-12 mt-5'>
                            <table className='table-basket'>
                                <tbody>
                                    {phones.map((phone, index) => this.renderPhone(phone, index))}
                                </tbody>
                            </table>
                        </div>
                        <Modal
                            visible={this.state.modalEditVisible}
                            onOk={() => this.setModalEditVisible(false)}
                            onCancel={() => this.setModalEditVisible(false)}
                            phones={phones}
                            editPhoneId={editPhoneId}
                            history={this.props.history}
                        />
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state, ownProps) => ({
    phones: getPhones(state, ownProps),
});

const mapDispatchToProps = {
    fetchPhones,
    editPhoneId
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Admin)
